### PR TITLE
[READY] Chainsaw and a tree

### LINKED ISSUE #23704: Chainsaw and a tree.
Game version:
current experimental

### Expected behavior
Cutting a tree with chainsaw should be reasonably quicker than with an axe.

### Actual behavior
It is not. Current formula in source code:

`moves = MINUTES( 70 - p->str_cur ) / it->get_quality( AXE ) * 100;`

So, if we assume 10 STR character, times are:
primitive axe (q1): 60m
wood axe (q2): 30m
chainsaw (q3): 20m

Not worth the gas/batteries, if you ask me. Also, 20 minutes to fell a tree with chainsaw is ridiculous.

### Suggested solution
I just checked the video on YT, and it should be more like
5min flat for safety and cleaning preparations and 20s (for chainsaw) to do the cutting.

Same issue with cutting trunk into the logs.

### PR DESCRIPTION
**Pull request by Robik81**
Closes #23704 

As I noted in issue mentioned above, tree cutting with chainsaw took way too long.

To solve this isse, I used formula (60-STR) / (2^(AXE-1) ) suggested by Maddremor. One small difference is that for powered tools it uses DEX instead of STR.

For person with STR / DEX 8:
Q1 - 52m
Q2 - 26m
Q3 - 13m - moved combat chainsaw here (from Q1)
Q4 - 6m - moved chainsaw here (from Q3)

Times for various tool quality and attributes is nicely presented in Maddremor post in the issue thread.

### CODE REVIEW

File: src/iuse.cpp [hunk 1]
```diff
@@ -4109,8 +4121,7 @@ int iuse::chop_logs( player *p, item *it, bool t, const tripoint &pos )
 
     const ter_id ter = g->m.ter( dirp );
     if( ter == t_trunk ) {
-        /** @EFFECT_STR reduces time required to chop down a tree */
-        moves = MINUTES( 70 - p->str_cur ) * 2 / it->get_quality( AXE ) * 100;
+        moves =  chop_moves( p, it );
```
Comment (DracoGriffin) [inline src/iuse.cpp]: extra space here between = and `chop_moves`

**Comment by Robik81** [comment]
> From testing, it looks like there is a variable range --- it's not a static cost. For default 8 stats survivor with either chainsaw / electric chainsaw, the time it takes seems to be anywhere around 5m30s~ to 7m15s~. Survivor with all 20 stats with same chainsaws, was doing consistently around 5m~ to 5m15s~.

Possible, but the variability is IMO introduced somewhere in processing of the long action and not in the code I changed. Change of character speed due to bad weather or something? I don't really know, but I bet if there is variability now, it was there before.

**Comment by Robik81** [comment]
> Giving combat chainsaw Q3 seems inappropriate, especially with the description as follows, emphasis mine:
> " "description": "This is a chainsaw that has been lightened, tuned, and extensively modified to be a more effective weapon. Unfortunately these modifications have rendered it much less effective as a woodcutting tool.","

Well, if your employer dropped your wage by 50%, would you consider it much less, or just a little bit less? ;-)

On the other hand, I have no problem changing combat chainsaw back, I never use it anyway... Could we have opinion of more people on this one?

**Comment by sfsworms** [comment]
I'm with robik. Twice the time is way worse than the chainsaw, and giving it woodcutting ii would only make it as good as a wood saw, which seems bad for a powered too

**Comment by RadHazard** [comment]
I suggest that the wood axe be bumped back up to axe 3 now as well to differentiate it from the other axes not built for woodcutting

**Comment by Robik81** [comment]
Okay, so far these are suggested changes.

* move combat chainsaw back to Q1
* move wood axe to Q3

I have no problem with either of these, but decision ultimately lies with people with merging rights.

Feel free to merge this one with necessary tuning or just post balancing changes that need to be done before merging and I will commit them into the PR.

**Pull request merged**